bears/installation_tool.py

```diff
@@ -0,0 +1,158 @@
+from pygments.lexers import SqlLexer
+
+
+def install_pip_package(package_name):
+    """
+    Uses call to install a PyPi package.
+
+    :param package_name: The package to be installed.
+    """
+    print('To install the dependencies correctly, we will need administrator '
+          'rights from you.')
+    subprocess.call(['sudo', sys.executable, '-m',
+                     'pip', 'install', package_name, '--upgrade'])
+
+
+def get_output(command):
+    """
+    Runs the command and decodes the output and returns it.
+
+    :param command: The command to be run.
+    :return:        The output of the command, decoded.
+    """
+    function = subprocess.Popen(command, stdout=subprocess.PIPE)
+    output, err = function.communicate()
+    return output.decode("utf-8")
+
+
+def check_failed_bears(bear_list):
+    """
+    Checks if any bear failed installation. Exits with 1 if so.
+    :param bear_list: List of the failed bears.
+    """
+    if bear_list:
+        print('Bears that failed installing:')
+        for bear_failed in bear_list:
+            print(bear_failed)
+        sys.exit(1)
+
+
+def install_requirements(package_name, package_failed_list):
+    """
+    Imports a package and tries installing its requirements.
+
+    :param package_name:        The package to be imported.
+    :param package_failed_list: The list with the packages which their
+                                requirements failed installing.
+    :return:                    A list with the packages which had their
+                                requirements failing to be installed.
+    """
+    package_object = importlib.import_module(
+        package_name + "." + package_name)
+    for requirement in getattr(package_object, package_name).REQUIREMENTS:
+        try:
+            subprocess.call([requirement.install_command()])
+        except:
+            package_failed_list.append(package_name)
+    return package_failed_list
+
+
+def get_all_bears_names_from_PyPI():
+    """
+    Gets all the bears names from PyPI, using the link in the description.
+
+    :return: A list with all the bear names.
+    """
+    output = get_output(['pip', 'search', "coala.rtfd.org"])
+    return re.findall(r'(.+) \(\d.*\n', output)
+
+
+def install_bears(bear_names_list, arg):
+    """
+    Tries to install each bear from the bear_names_list. Will also check for
+    bears which failed to be installed, or their requirements failed to be
+    installed.
+
+    :param bear_names_list: The list which contains the names of the bears.
+    :param arg:             An arg which is given to also install the bears'
+                            dependencies.
+    """
+    for bear_name in bear_names_list:
+        bear_name = bear_name.replace(' ', '')
+        bears_failed_list = []
+        try:
+            install_pip_package(bear_name)
+        except:
+            bears_failed_list.append(bear_name)
+
+        if arg:
+            bears_failed_list = install_requirements(
```

Comment:
Would rather not give`bears_failed_list` as a param to `install_requirements` . Instead let it build its own failure list and then you do: `bears_failed_list += install_requirements(bear_name)`